awx/ui/src/components/AppContainer/NavExpandableGroup.js

```diff
@@ -1,60 +1,47 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes, { oneOfType, string, arrayOf } from 'prop-types';
-import { matchPath, Link, withRouter } from 'react-router-dom';
+import { matchPath, Link, useHistory } from 'react-router-dom';
 import { NavExpandable, NavItem } from '@patternfly/react-core';
 
-class NavExpandableGroup extends Component {
-  constructor(props) {
-    super(props);
-    const { routes } = this.props;
+function NavExpandableGroup(props) {
+  const history = useHistory();
+  const { groupId, groupTitle, routes } = props;
 
-    // Extract a list of paths from the route params and store them for later. This creates
-    // an array of url paths associated with any NavItem component rendered by this component.
-    this.navItemPaths = routes.map(({ path }) => path);
-    this.isActiveGroup = this.isActiveGroup.bind(this);
-    this.isActivePath = this.isActivePath.bind(this);
-  }
+  // Extract a list of paths from the route params and store them for later. This creates
+  // an array of url paths associated with any NavItem component rendered by this component.
+  const navItemPaths = routes.map(({ path }) => path);
 
-  isActiveGroup() {
-    return this.navItemPaths.some(this.isActivePath);
-  }
+  const isActive = () => {
+    navItemPaths.some(isActivePath);
+  };
 
-  isActivePath(path) {
-    const { history } = this.props;
+  function isActivePath(path) {
     return Boolean(matchPath(history.location.pathname, { path }));
   }
 
-  render() {
-    const { groupId, groupTitle, routes } = this.props;
-
-    if (routes.length === 1 && groupId === 'settings') {
-      const [{ path }] = routes;
-      return (
-        <NavItem itemId={groupId} isActive={this.isActivePath(path)} key={path}>
-          <Link to={path}>{groupTitle}</Link>
-        </NavItem>
-      );
-    }
-
+  if (routes.length === 1 && groupId === 'settings') {
+    const [{ path }] = routes;
     return (
-      <NavExpandable
-        isActive={this.isActiveGroup()}
-        isExpanded
-        groupId={groupId}
-        title={groupTitle}
-      >
-        {routes.map(({ path, title }) => (
-          <NavItem
-            groupId={groupId}
-            isActive={this.isActivePath(path)}
-            key={path}
-          >
-            <Link to={path}>{title}</Link>
-          </NavItem>
-        ))}
-      </NavExpandable>
+      <NavItem itemId={groupId} isActive={isActivePath(path)} key={path}>
+        <Link to={path}>{groupTitle}</Link>
+      </NavItem>
     );
   }
+
+  return (
+    <NavExpandable
+      isActive={isActive()}
```

Comment:
If you make `isActive` a variable, you only need to pass it and not call it here.